views/layouts/MainLayout.jsx

```diff
@@ -1,52 +1,85 @@
 import React, { useState, useEffect, useContext } from 'react'
-import { Outlet, useLocation, useNavigate } from 'react-router-dom'
+import { Outlet, useLocation, useNavigate, useParams } from 'react-router-dom'
 import classNames from 'classnames'
 import Header from '../components/Header'
 import Sidebar from '../components/Sidebar'
 
 import api from '../api'
 import getCounts from '../fe-utils/countUtil'
 import { fetchSubjects } from '../fe-utils/fetchUtil'
-import { AuthContext, NotificationContext } from '../contexts'
+import {
+  AuthContext,
+  ConfigurationsContext,
+  NotificationContext,
+  SidebarContext,
+} from '../contexts'
 import { headerTitle } from './helpers'
 
+const TEMPORARY_SIDEBAR = 'temporary'
+const PERSISTENT_SIDEBAR = 'persistent'
+const dashboard = 'dashboard'
+
 const MainLayout = ({ classes, theme }) => {
+  const [configurations, setConfigurations] = useContext(ConfigurationsContext)
   const [, setNotification] = useContext(NotificationContext)
+  const [openSidebar, setOpenSidebar] = useContext(SidebarContext)
   const [user, setUser] = useContext(AuthContext)
   const [users, setUsers] = useState([])
+  const [persistSidebar, setPersistSidebar] = useState(PERSISTENT_SIDEBAR)
   const { pathname } = useLocation()
+  const params = useParams()
   const navigate = useNavigate()
-
-  const [openDrawer, setOpenDrawer] = useState(false)
   const [sideBarState, setSideBarState] = useState({
     totalDays: 0,
     totalStudies: 0,
     totalSubjects: 0,
   })
-
-  const toggleDrawer = () => setOpenDrawer(!openDrawer)
+  const toggleSidebar = () => setOpenSidebar(!openSidebar)
   const fetchUsers = async () => {
     const usersList = await api.users.loadAll()
     setUsers(usersList)
   }
+  const loadAllConfigurations = async (userId) => {
+    try {
+      const configurations = await api.userConfigurations.all(userId)
```

Comment:
since we're only getting the current user's configurations we could remove the `userId` input here.

```suggestion
  const loadAllConfigurations = async () => {
    try {
      const configurations = await api.userConfigurations.all(user.uid)
```

Reply:
e918d142a1011200ad2eead272cfaea47ef32cdc